src/core/query.ts

```diff
@@ -412,6 +413,17 @@ export class Query<
       meta: this.meta,
     }
 
+    Object.defineProperty(context, 'signal', {
+      enumerable: true,
+      get: () => {
+        if (abortController) {
+          this.abortSignalConsumed = true
+          return abortController.signal
+        }
+        return undefined
+      },
```

Comment:
is there a way to extract this function into a const so that we can use the same getter for both signals?

Reply:
The whole `Object.defineProperty` was exactly the same, so I moved the whole thing to a separate function used in both of the contexts.